sklearn/discriminant_analysis.py

```diff
@@ -443,11 +446,24 @@ def fit(self, X, y):
             self.priors_ = self.priors_ / self.priors_.sum()
 
         # Get the maximum number of components
+
+        # Maximum number of components no matter what n_components is
+        # specified:
+        max_components = min(len(self.classes_) - 1, X.shape[1])
+
         if self.n_components is None:
-            self._max_components = len(self.classes_) - 1
+            self._max_components = max_components
```

Comment:
I feel the private variable should be called ``_n_components``, not ``_max_components``.

Reply:
I agree, however maybe it could be justified to call it `_max_components` for the case where some inputs of LDA would be collinear ? indeed the `scalings_` would be truncated and have less components than `_n_components` (see issue #11528)